src/transform/pre_reduce.rs

```diff
@@ -19,39 +22,131 @@ impl run::AsDef for InertDef {
   }
 }
 
+pub struct PreReduceStats {
+  pub rewrites: Rewrites,
+  pub errors: Vec<String>,
```

Comment:
this is unused afaict

Reply:
I meant the `errors` field – which is always just `vec![]`

Reply:
I wanted to have a place in the public API to put potential errors in (though right now, we can't detect OOM errors, for example). Do you think this is necessary, or is it premature right now?

Reply:
I think it's premature for now. (For example, we might want to use an error enum instead of `String`.)